aviatrix/data_source_aviatrix_all_transit_gateways.go

```diff
@@ -0,0 +1,870 @@
+						},
+						"enable_spot_instance": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Enable spot instance. NOT supported for production deployment.",
+						},
+						"spot_price": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Price for spot instance. NOT supported for production deployment.",
+						},
+						"azure_eip_name_resource_group": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the public IP address and its resource group in Azure to assign to this Transit Gateway.",
+						},
+						"ha_azure_eip_name_resource_group": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the public IP address and its resource group in Azure to assign to the HA Transit Gateway.",
+						},
+						"local_as_number": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Changes the Aviatrix Transit Gateway ASN number before you setup Aviatrix Transit Gateway connection configurations.",
+						},
+						"ha_security_group_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "HA security group used for the transit gateway.",
+						},
+						"lan_interface_cidr": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Transit gateway lan interface cidr.",
+						},
+						"ha_lan_interface_cidr": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Transit gateway lan interface cidr for the HA gateway.",
+						},
+						"bgp_lan_ip_list": {
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Computed: true,
+							Description: "List of available BGP LAN interface IPs for transit external device connection creation. " +
+								"Only supports GCP. Available as of provider version R2.21.0+.",
+						},
+						"ha_bgp_lan_ip_list": {
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Computed: true,
+							Description: "List of available BGP LAN interface IPs for transit external device HA connection creation. " +
+								"Only supports GCP. Available as of provider version R2.21.0+.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAviatrixAllTransitGatewayRead(d *schema.ResourceData, meta interface{}) error {
+
+	if !d.Get("all_transit_gateway").(bool) {
+		return nil
+	}
+
+	client := meta.(*goaviatrix.Client)
+
+	TransitGatewayNameList, err := client.GetTransitGatewayNameList()
+	if err != nil {
+		return err
+	}
+	var result []map[string]interface{}
+	for i := range TransitGatewayNameList {
+		transitGatewayName := TransitGatewayNameList[i]
+		gateway := &goaviatrix.Gateway{
+			GwName: transitGatewayName,
+		}
+		gw, err := client.GetGateway(gateway)
```

Comment:
This isn't right, you shouldn't be calling an API for every single gateway as this resource will take too long to read for a large number of gateways. If you call `list_vpcs_summary` without the `gateway_name` attribute, it will return information for all gateways, which should be better than calling `list_vpcs_summary` each time for every gateway. It looks like you are already doing that when you call `GetTransitGatewayNameList`. Just use that information instead of filtering only the gateway names out.

Also, when calling `list_vpcs_summary` without `gateway_name`, some attributes will not be returned, which is ok. You should just remove those from the data source.

Reply:
have fixed, currently will use the GetTransitGatewayList() func, which returns a list including the all transit gateways, by this way only call on time.